Create 11.3.0 release notes (#259)
giantswarm/releases: release-notes/azure/v11.3.0.md

## :zap:  Giant Swarm Release 11.3.0 for Azure :zap:

This release replaces CoreOS Container Linux that is going [end of life](https://coreos.com/os/eol/) on 26th of May 2020 with Flatcar Container Linux. 

> Migration process requires replacing VMSS CoreOS image with Flatcar image which, due to Azure limitations, is not possible. In order to enable existing clusters to upgrade with this release we had to amend standard procedure with additional steps that replace old VMSS with a new one, ensuring all workload is moved. 

Reviewer: ```suggestion
Migration process requires replacing VMSS CoreOS image with Flatcar image which, [due to Azure limitations](https://docs.microsoft.com/en-us/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-upgrade-scale-set#create-time-properties), is not possible. In order to enable existing clusters to upgrade with this release we had to amend standard procedure with additional steps that replace old VMSS with a new one, ensuring all workload is moved. 
```

### Warning
Due to additional steps in upgrade process, the whole procedure will take more time, up to a few hours. It also requires manual steps on our side, so please **schedule all upgrades with your SE at least one day in advance**. This will allow us to prepare and guarantee appropriate support throughout the upgrade process.

> Initial tests shown up to 2% loss of incoming traffic over the time period of 5-10min. It was not confirmed in the following tests but because of the complexity of the process you should be aware that this may occur.

Reviewer: is this still true? During standup I understood we had 100% uptime, but I may have been distracted doing memes.

Author: I think after fixing @whites11 still observed 2% drop which was better than first 25% but it is correct that @tuommaki had 100% uptime. I wasn't sure if it was something you changed @tuommaki or your tests didn't show any downtime?

Reviewer 2: Let's not say it is 100% uptime as they will complain on that for sure :D

It is important to remember that while the switch is in progress for masters, the workloads will be working as long as they do not fail, as the rescheduling will not be possible.
3. When new VMSS master instance is ready, upgrade will be waiting for the GS staff to migrate the ETCD data manually from old VMSS to the new instance. After the migration is done, the old instance is removed and new instance will join the cluster.
4. After masters are done, we will move to worker instances. Here the process is simpler, where we create a second VMSS with flatcar images and we will be migrating workloads as it was done during previous upgrades.
5. Finally when all workloads are migrated, we will delete old workers.

Due to additional steps in upgrade process, the whole procedure will take more time. It also requires manual steps on our side, so please **schedule all upgrades with your SE at least one day in advance**. This will allow us to prepare and guarantee appropriate support throughout the upgrade process.

> Initial tests shown up to 2% loss of incoming traffic over the time period of 5-10min. It was not confirmed in the following tests but because of the complexity of the process you should be aware that this may occur.

Reviewer: My english is worse than yours but doesn't it sound better like this?

```suggestion
Initial tests shown up to 2% loss of incoming traffic over the time period of 5-10min. It was not reproduced in the following tests but because of the complexity of the process you should be aware that this may occur.
```

## :zap:  Giant Swarm Release 11.3.0 for Azure :zap:

> This release replaces CoreOS Container Linux that is going [end of life](https://coreos.com/os/eol/) on 26th of May 2020 with Flatcar Container Linux. 

Reviewer 3: ```suggestion
This release replaces CoreOS Container Linux that is going [end of life](https://coreos.com/os/eol/) on the 26th of May 2020 with Flatcar Container Linux. 
```

**The creation of new clusters with this release works as usual. New clusters will contain Flatcar OS from the start.**

> Flatcar image on the contrary of the CoreOS has a licence that has to be accepted. Please look into the instructions prior to creating or upgrading your clusters [here](https://github.com/giantswarm/releases/blob/master/announcements/11-3-0-azure-flatcar-licence-approval.md)

Reviewer 3: ```suggestion
Flatcar Container Linux images, contrary to the CoreOS Container Linux image, is under a license which you need to accept. Please look into the instructions prior to creating or upgrading your clusters [here](https://github.com/giantswarm/releases/blob/master/announcements/11-3-0-azure-flatcar-licence-approval.md)
```

> **The creation of new clusters with this release works as usual, no changes has been made here.**

Author: ```suggestion
**The creation of new clusters with this release works as usual. New clusters will contain Flatcar OS from the start.**
```

> Migration process requires replacing VMSS CoreOS image with Flatcar image which, [due to Azure limitations](https://docs.microsoft.com/en-us/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-upgrade-scale-set#create-time-properties), is not possible. In order to enable existing clusters to upgrade with this release we had to amend standard procedure with additional steps that replace old VMSS with a new one, ensuring all workload is moved. 

Reviewer 3: ```suggestion
The migration process requires replacing the VMSS CoreOS Container Linux image with Flatcar Container Linux image which, [due to Azure limitations](https://docs.microsoft.com/en-us/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-upgrade-scale-set#create-time-properties), is not possible in-place. In order to enable existing clusters to upgrade to this release, we had to amend the standard procedure with additional steps to replace the old VMSS with a new one, ensuring all your workload is moved.
```

> Flatcar image on the contrary of the CoreOS has a licence that has to be accepted. Please look into the instructions prior to upgrading your clusters [here](https://github.com/giantswarm/releases/blob/master/announcements/11-3-0-azure-flatcar-licence-approval.md)

Reviewer 3: ```suggestion
Flatcar Container Linux image,  contrary to the CoreOS Container Linux image, has a licence that has to be accepted. Please look into the instructions prior to upgrading your clusters [here](https://github.com/giantswarm/releases/blob/master/announcements/11-3-0-azure-flatcar-licence-approval.md)
```

### Warning

> Following steps will be undertaken during the upgrade procedure:

Reviewer 3: ```suggestion
The following steps will be undertaken during the upgrade procedure:
```

1. Once you start the upgrade, we will be notified in order to be hands on during the procedure.

> 2. Upgrade will start with master replacement. Here old master is stopped as we need a final state of the ETCD data for the migration. Then new VMSS master instance is created (which due to Azure recent performance with disk provisioning can take several tries, hence extend the upgrade time). 

Reviewer 3: ```suggestion
2. The upgrade will start by replacing the master. The old master will be stopped as we need the final state of the ETCD data to be frozen for the migration. Once this is done, the new VMSS for the master instance will be created (:information_source: Due to the recent Azure disk provisioning performance,  this can take several tries, hence extend the upgrade time). 
```

> It is important to remember that while the switch is in progress for masters, the workloads will be working as long as they do not fail, as the rescheduling will not be possible.

Reviewer 3: ```suggestion
It is important to remember that while the switch is in progress for the master instance, the workloads will be unaffected as long as the pods do not fail, as rescheduling will not be possible as long as the master is not up.
```

> 3. When new VMSS master instance is ready, upgrade will be waiting for the GS staff to migrate the ETCD data manually from old VMSS to the new instance. After the migration is done, the old instance is removed and new instance will join the cluster.

Reviewer 3: ```suggestion
3. When the new VMSS master instance is ready, yhr upgrade will be waiting for the GS staff to migrate the ETCD data manually from the old master instance to the new one. After the migration is done on our side, the old instance will be removed and the new instance will join the cluster.
```

Reviewer 2: You made a typo here, can you change it @omegas27 ?

Reviewer 3: Fixed :)

> 4. After masters are done, we will move to worker instances. Here the process is simpler: we create VMSS with flatcar images and migrate workloads, same as it was done during previous upgrades.

Reviewer 3: ```suggestion
4. After the master is upgraded successfully, the upgrade will proceed with the worker instances. Here, the process is simpler. The Azure Operator will create a VMSS with Flatcar Container Linux images and migrate the workloads, as you are used to during the previous upgrades.
```

> 5. Finally when all workloads are migrated, we will delete old workers.

Reviewer 3: ```suggestion
5. Finally, when all your workloads is migrated, the VMSS running the CoreOS Container Linux instances will be deleted.
```

Reviewer 2: if `workloads` then are? it is plural?

Reviewer 3: Solved

> Due to additional steps in upgrade process, the whole procedure will take more time. It also requires manual steps on our side, so please **schedule all upgrades with your SE at least one day in advance**. This will allow us to prepare and guarantee appropriate support throughout the upgrade process.

Reviewer 3: ```suggestion
Due to the additional steps in the upgrade process, the whole procedure will take more time. As it also requires a manual intervention on our side, could you please **schedule all the upgrades to 11.3.0 with your Solution Engineer at least a day in advance**. This will allow us to prepare and guarantee you the appropriate support throughout the upgrade process.
```

> Initial tests shown up to 2% loss of incoming traffic over the time period of 5-10min. It was not reproduced in the following tests but because of the complexity of the process you should be aware that this may occur.

Reviewer 3: ```suggestion
Our initial tests showed up to 2% loss of incoming traffic over the time period of 5-10min. It was not reproduced in the following tests but because of the complexity of the process you should be aware that this may occur.
```
